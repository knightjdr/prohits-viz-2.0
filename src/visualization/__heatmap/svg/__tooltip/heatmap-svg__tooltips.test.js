import React from 'react';
import { shallow } from 'enzyme';

import Tooltips from './heatmap-svg__tooltips';

const clearTooltip = jest.fn();
const handleMouseMove = jest.fn();

describe('Tooltips', () => {
  describe('with mouse events', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Tooltips
          clearTooltip={clearTooltip}
          handleMouseMove={handleMouseMove}
          height={100}
          mouseEvents
          setRef={{ ref: 'test' }}
          width={200}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should set pointer events', () => {
      expect(wrapper.find('g').props().pointerEvents).toBe('auto');
    });

    describe('rect listener', () => {
      let rect;

      beforeAll(() => {
        rect = wrapper.find('rect');
      });

      it('should set height', () => {
        expect(rect.props().height).toBe(100);
      });

      it('should set width', () => {
        expect(rect.props().width).toBe(200);
      });

      it('should handle mouse leave', () => {
        clearTooltip.mockClear();
        rect.simulate('mouseleave');
        expect(clearTooltip).toHaveBeenCalled();
      });

      it('should handle mouse mouse', () => {
        handleMouseMove.mockClear();
        rect.simulate('mousemove');
        expect(handleMouseMove).toHaveBeenCalled();
      });
    });
  });

  describe('without mouse events', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Tooltips
          clearTooltip={clearTooltip}
          handleMouseMove={handleMouseMove}
          height={100}
          mouseEvents={false}
          setRef={{ ref: 'test' }}
          width={200}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should set pointer events', () => {
      expect(wrapper.find('g').props().pointerEvents).toBe('none');
    });
  });
});
