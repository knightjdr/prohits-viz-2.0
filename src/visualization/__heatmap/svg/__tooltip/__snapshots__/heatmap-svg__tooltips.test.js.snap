// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltips with mouse events should match snapshot 1`] = `
<React.Fragment>
  <g
    pointerEvents="auto"
    transform="translate(100 100)"
  >
    <rect
      fill="black"
      height={100}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      opacity="0"
      width={200}
      x="0"
      y="0"
    />
  </g>
</React.Fragment>
`;

exports[`Tooltips without mouse events should match snapshot 1`] = `
<React.Fragment>
  <g
    pointerEvents="none"
    transform="translate(100 100)"
  >
    <rect
      fill="black"
      height={100}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      opacity="0"
      width={200}
      x="0"
      y="0"
    />
  </g>
</React.Fragment>
`;
